### DISCUSSION TITLE
part-ii/2016-32/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#151] [Question comments]
Part II 2016 Coding and Cryptography: Paper 2, Section I, G 

https://questions.tripos.org/part-ii/2016-32/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

Suppose we have source $X$ and output $Y$, with $\mathbb{P}(X=0) = \alpha$, so $\mathbb{P}(X=1) = 1 - \alpha$, $\mathbb{P}(Y=0) = \frac{1}{2} + \frac{\alpha}{2}$, $\mathbb{P}(Y=1) = \frac{1}{2} - \frac{\alpha}{2}$.

The capacity is given by

$$
\begin{align*}
C &= \max_{\alpha} H(Y) - H(Y|X) \\
&= \max_{\alpha} [H(\frac{1}{2}\alpha + \frac{1}{2}) + (\alpha - 1)]
\end{align*}
$$

Differentiating to find the maximum gives

$$\frac{1}{2}\log{(\frac{\frac{1}{2} - \frac{1}{2}\alpha}{\frac{1}{2} + \frac{1}{2}\alpha})} + 1 = 0$$

which gives $\alpha = \frac{3}{5}$. Therefore

$$
\begin{align*}
C &= H(\frac{4}{5}) - \frac{2}{5} \\
&= -\frac{4}{5}\log{4} + \frac{4}{5}\log{5} + \frac{1}{5}\log{5} - \frac{2}{5} \\
&= \log{5} - 2 \\
\end{align*}
$$

as required.

</details>
